lib/response.js

```diff
@@ -70,18 +70,43 @@ res.status = function status(code) {
  *      last: 'http://api.example.com/users?page=5'
  *    });
  *
- * @param {Object} links
+ *    res.links([{
+ *          href: 'http://api.example.com/users?page=2',
+ *          rel: 'next',
+ *          title: 'next chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }, {
+ *          href: 'http://api.example.com/users?page=5',
+ *          rel: 'last',
+ *          title: 'last chapter',
+ *          type: 'text/plain;charset=UTF-8'
+ *      }]);
+ *
+ *
+ * @param {Object|Array} links
  * @return {ServerResponse}
  * @public
  */
 
-res.links = function(links){
+res.links = function(links) {
   var link = this.get('Link') || '';
   if (link) link += ', ';
-  return this.set('Link', link + Object.keys(links).map(function(rel){
-    return '<' + links[rel] + '>; rel="' + rel + '"';
-  }).join(', '));
-};
+  if (Array.isArray(links)) { 
+    var attrTest = /^[a-z][a-z-*]+$/;
```

Comment:
``/^[\w!#$&+.^`|~-]+\*?$/`` would be the exact same as the productions in [rfc5988](https://tools.ietf.org/html/rfc5988) (see the link-extension production)